src/cloudsc_python/drivers/cloudsc_f2py.py

```diff
@@ -0,0 +1,144 @@
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+import click
+from pathlib import Path
+import numpy as np
+
+
+def loki_generate_kernel(source_path, out_path, include_dir=None):
+    from loki import Sourcefile, flatten
+    from loki.transform import FortranPythonTransformation
+
+    source_dir = source_path.parent
+    headers = ['yoethf.F90', 'yoecldp.F90', 'yomcst.F90']
+    definitions = flatten(
+        Sourcefile.from_file(source_dir/header).modules for header in headers
+    )
+
+    f2py = FortranPythonTransformation(
+        with_dace=False, suffix='_py', invert_indices=True
+    )
+
+    # Parse original driver and kernel routine, and enrich the driver
+    kernel = Sourcefile.from_file(
+        source_path, definitions=definitions,
+        includes=source_dir/'include', preprocess=True
+    )
+    f2py.apply(kernel, role='kernel', path=out_path)
+
+
+def cloudsc_validate(fields, ref_fields, kidia, kfdia):
+    _field_names = [
+        'plude', 'pcovptot', 'prainfrac_toprfz', 'pfsqlf', 'pfsqif',
+        'pfcqlng', 'pfcqnng', 'pfsqrf', 'pfsqsf', 'pfcqrng', 'pfcqsng',
+        'pfsqltur', 'pfsqitur', 'pfplsl', 'pfplsn', 'pfhpsl', 'pfhpsn',
+        'tendency_loc_a', 'tendency_loc_q', 'tendency_loc_t',
+        'tendency_loc_cld'
+    ]
+    ngptot = kfdia - kidia + 1
+
+    print(
+        "             Variable Dim             MinValue             MaxValue"
+        "            AbsMaxErr         AvgAbsErr/GP          MaxRelErr-%"
+    )
+    for name in _field_names:
+        if len(fields[name].shape) == 1:
+            f = fields[name][kidia-1:kfdia]
+            ref = ref_fields[name][kidia-1:kfdia]
+        elif len(fields[name].shape) == 2:
+            f = fields[name][:,kidia-1:kfdia]
+            ref = ref_fields[name][:,kidia-1:kfdia]
+        elif len(fields[name].shape) == 3:
+            f = fields[name][:,:,kidia-1:kfdia]
+            ref = ref_fields[name][:,:,kidia-1:kfdia]
+        else:
+            f = fields[name]
+            ref = ref_fields[name]
+        zsum = np.sum(np.absolute(ref))
+        zerrsum = np.sum(np.absolute(f - ref))
+        zeps = np.finfo(np.float64).eps
+        print(
+            ' {fname:>20}     {fmin:20.13e}  {fmax:20.13e}  {absmax:20.13e} '
+            ' {absavg:20.13e}  {maxrel:20.13e}'.format(
+                fname=name.upper(), fmin=f.min(), fmax=f.max(),
+                absmax=np.absolute(f - ref).max(),
+                absavg=np.sum(np.absolute(f - ref)) / ngptot,
+                maxrel=0.0 if zerrsum < zeps else (zerrsum/(1.0+zsum) if zsum < zeps else zerrsum/zsum)
+            )
+        )
+
+
+def run_cloudsc_kernel(nthreads, ngptot, nproma, input_path, reference_path):
```

Comment:
Am I missing something here or is this missing the input field expansion? Which means, any value for `ngptot` > 100 will not work?
Also, `nthreads` is not used (yet?).